From garzik@havoc.gtf.org Sun Oct  1 04:24:35 2006
Date: Sun, 1 Oct 2006 07:24:33 -0400
From: Jeff Garzik <jeff@garzik.org>
To: Greg KH <greg@kroah.com>
Cc: Andrew Morton <akpm@osdl.org>
Subject: usb/serial/mos7840: fix cast
Message-ID: <20061001112433.GA26855@havoc.gtf.org>
Content-Disposition: inline


Debug code in mos7840 emits the following warnings on 64-bit:

drivers/usb/serial/mos7840.c: In function "mos7840_open"
drivers/usb/serial/mos7840.c:1091: warning: cast from pointer to integer of different size
drivers/usb/serial/mos7840.c:1091: warning: cast from pointer to integer of different size
drivers/usb/serial/mos7840.c:1091: warning: cast from pointer to integer of different size

Solution:  Don't assume pointers are 32-bit.  The easy solution actually
cleans up the code:  "%p" (pointer) permits us to eliminate the casts,
and print correct debug information on 64-bit.

Signed-off-by: Jeff Garzik <jeff@garzik.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/mos7840.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/serial/mos7840.c
+++ gregkh-2.6/drivers/usb/serial/mos7840.c
@@ -1087,7 +1087,7 @@ static int mos7840_open(struct usb_seria
 	mos7840_port->icount.tx = 0;
 	mos7840_port->icount.rx = 0;
 
-	dbg("\n\nusb_serial serial:%x       mos7840_port:%x\n      usb_serial_port port:%x\n\n", (unsigned int)serial, (unsigned int)mos7840_port, (unsigned int)port);
+	dbg("\n\nusb_serial serial:%p       mos7840_port:%p\n      usb_serial_port port:%p\n\n", serial, mos7840_port, port);
 
 	return 0;
 
